libraries/networking/src/ResourceCache.cpp

```diff
@@ -402,7 +402,7 @@ void Resource::handleReplyFinished() {
                 const int MAX_ATTEMPTS = 8;
                 const int BASE_DELAY_MS = 1000;
                 if (++_attempts < MAX_ATTEMPTS) {
-                    QTimer::singleShot(BASE_DELAY_MS * (int)pow(2.0, _attempts), this, SLOT(attemptRequest()));
+                    QTimer::singleShot(BASE_DELAY_MS * (int)pow(2.0, _attempts), this, &Resource::makeRequest);
```

Comment:
The code here is pretty confusing, but I think it is correct. The calls to `requestCompleted` will increase the request limit. If we get rid of those and use a timer with makeRequest, we won't allow pending requests to be made in the interim.